miasm2/core/asmblock.py

```diff
@@ -359,132 +323,209 @@ class AsmSymbolPool(object):
+
+        loc_key = LocKey(self._loc_key_num)
+        self._loc_key_num += 1
+
+        self._loc_keys.add(loc_key)
+
+        if offset is not None:
+            assert offset not in self._offset_to_loc_key
+            self._offset_to_loc_key[offset] = loc_key
+            self._loc_key_to_offset[loc_key] = offset
+
+        self._name_to_loc_key[name] = loc_key
+        self._loc_key_to_name[loc_key] = name
+        return loc_key
+
+    def remove_location(self, loc_key):
         """
-        Delete a @label
+        Delete a @loc_key
         """
-        self._name2label.pop(label.name, None)
-        self._offset2label.pop(label.offset, None)
-        if label in self._labels:
-            self._labels.remove(label)
+        name = self._loc_key_to_name.pop(loc_key, None)
+        _ = self._name_to_loc_key.pop(name, None)
+
+        offset = self._loc_key_to_offset.pop(loc_key, None)
+        _ = self._offset_to_loc_key.pop(offset, None)
+
+        self._loc_keys.remove(loc_key)
 
-    def del_label_offset(self, label):
-        """Unpin the @label from its offset"""
-        self._offset2label.pop(label.offset, None)
-        label.offset = None
+    def del_loc_key_offset(self, loc_key):
+        """Unpin the @loc_key from its offset"""
+        offset = self._loc_keys_to_offset.pop(loc_key)
+        _ = self._offset_to_loc_key.pop(offset, None)
 
     def getby_offset(self, offset):
-        """Retrieve label using its @offset"""
-        return self._offset2label.get(offset, None)
+        """Retrieve loc_key using its @offset"""
+        return self._offset_to_loc_key.get(offset, None)
 
     def getby_name(self, name):
-        """Retrieve label using its @name"""
-        return self._name2label.get(name, None)
+        """Retrieve loc_key using its @name"""
+        return self._name_to_loc_key.get(name, None)
 
     def getby_name_create(self, name):
-        """Get a label from its @name, create it if it doesn't exist"""
-        label = self.getby_name(name)
-        if label is None:
-            label = self.add_label(name)
-        return label
+        """Get a loc_key from its @name, create it if it doesn't exist"""
+        loc_key = self.getby_name(name)
+        if loc_key is None:
+            loc_key = self.add_location(name)
+        return loc_key
 
     def getby_offset_create(self, offset):
-        """Get a label from its @offset, create it if it doesn't exist"""
-        label = self.getby_offset(offset)
-        if label is None:
-            label = self.add_label(offset, offset)
-        return label
-
-    def rename_label(self, label, newname):
-        """Rename the @label name to @newname"""
-        if newname in self._name2label:
+        """Get a loc_key from its @offset, create it if it doesn't exist"""
+        loc_key = self.getby_offset(offset)
+        if loc_key is None:
+            loc_key = self.add_location(offset)
+        return loc_key
+
+    def rename_location(self, loc_key, newname):
```

Comment:
For homogeneity, this could be `set_name` and just set a new name to a `loc_key`, which may not have one already.

Reply:
See with @commial: not a bug as name is mandatory